subseas_flex_fcst/maproom.py

```diff
@@ -45,26 +48,75 @@
 
 APP.layout = layout.app_layout
 
-
-def read_cptdataset(y_transform=False):
-
-    fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+#function to open all datasets and then combine them
+def combine_cptdataset(dataDir,filePattern,dof=False,y_transform=False):
+    filesNameList = glob.glob(f'{dataDir}/{filePattern}')
+    dataDic = {}
+    for idx, i in enumerate(filesNameList):
+        fileCompList = re.split('[_.]',filesNameList[idx])
+        fileCompList = [x for x in fileCompList if "/" not in x]
+
+        leadTimeWk = [x for x in fileCompList if x.startswith('wk')]
+
+        dateFormat = re.compile(".*-.*-.*")
+        startDateStr = list(filter(dateFormat.match,fileCompList))
+        startDate = startDateStr[0]
+        #startDate = datetime.strptime(startDateStr[0],
+        #'%b-%d-%Y').date()
+        if leadTimeWk[0] == 'wk1':
+            leadTimeDate = 'Week 1'#'Apr 2' #should it be made into dates or kept as wk number?
```

Comment:
Why don't you just use the numeric lead times as the values of L, rather than using strings here and then converting them to numbers later?

Reply:
I did this based off of how the data was given to me; I figured we would want to keep it as close to the way the original data was expressed. But I can change this since, based off the way to email thread is going, it seems like we might want to express S,L in a more mathematically easeful way within the code over trying to stick to user convenience / similarity to the metadata?